src/dataregistry/registrar/keyword.py

```diff
@@ -0,0 +1,282 @@
+    It provides methods to add, remove, and list keywords associated with datasets.
+
+    Parameters
+    ----------
+    db_connection : DbConnection object
+        Encompasses sqlalchemy engine, dialect (database backend)
+        and schema version
+    owner : str
+        To set the default owner for all registered keywords in this
+        instance.
+    """
+    def __init__(
+            self,
+            db_connection: DbConnection,
+            root_dir,
+            owner,
+            owner_type
+    ) -> None:
+        super().__init__(db_connection, root_dir, owner, owner_type)
+        self.which_table = "keyword"
+        self.entry_id = "keyword"
+
+    def create_keyword(
+            self,
+            keyword: str,
+            user_type: Literal["user", "group", "project"] = "user",
+            system: bool = False,
+            commit: bool = True):
+        """
+        Add a keyword to the registry.
+
+        Parameters
+        ----------
+        keyword : str
+            The keyword to add.
+        """
+        owner = self._owner or os.getenv("USER")
+        if not isinstance(keyword, str):
+            raise ValueError(f"Keyword {keyword} is not a valid keyword string.")
+        kwargs_dict = {"keyword": keyword.lower(),
+                       "creator_uid": owner,
+                       "system": system,
+                       "active": True,
+                       "creation_date": datetime.now()}
+        # Implementation for adding a keyword to the dataset in the database
+        keywords_table = self._get_table_metadata("keyword")
+        with self._engine.connect() as conn:
+            add_table_row(conn, keywords_table, kwargs_dict, commit=commit)
+
+    def create_keywords(
+            self,
+            keywords: list[str],
+            user_type: Literal["user", "group", "project"] = "user",
+            system: bool = False):
+        """
+        Add multiple keywords to the registry.
+
+        Parameters
+        ----------
+        keywords : list[str]
+            The keywords to add.
+        """
+        for keyword in keywords:
+            if not isinstance(keyword, str):
+                raise ValueError(f"Keyword {keyword} is not a valid keyword string.")
+        self.create_keyword(keyword, user_type=user_type, system=system, commit=False)
+        with self._engine.connect() as conn:
+            conn.commit()
+
+    def disable_keyword(self, keyword: str):
+        """
+        Disable a keyword from the registry.
+
+        keyword must exist and must be owned by the current user.
+
+        Parameters
+        ----------
+        keyword : str
+            The keyword to disable.
+        """
+        self._set_enable_keyword(keyword, enable=False)
+
+    def enable_keyword(self, keyword: str) -> None:
+        """
+        Enable a keyword in the registry.
+
+        Keyword must exist and must be owned by the current user.
```

Comment:
I'm not sure about that restriction - what happens when someone leaves the collaboration? - but for now let's leave it.